pkg/block/fetcher.go

```diff
@@ -603,11 +604,13 @@ func (f *DeduplicateFilter) filterForResolution(root *Node, metaSlice []*metadat
 
 	duplicateULIDs := getNonRootIDs(root)
 	for _, id := range duplicateULIDs {
+		f.mu.Lock()
 		if metas[id] != nil {
 			f.duplicateIDs = append(f.duplicateIDs, id)
 		}
 		synced.WithLabelValues(duplicateMeta).Inc()
 		delete(metas, id)
+		f.mu.Unlock()
```

Comment:
This could be right before `synced.WithLabelValues(duplicateMeta).Inc()`. I think it would be more clear the intention, given we're not trying to synchronise on `metas`, just on `duplicateIDs`.

Reply:
hmm I am not sure I agree. The map that we need to synchronize is `metas`. `duplicateIDs`  is a slice and is not the source of the panic `fatal error: concurrent map writes`

Reply:
Also, note that we need the lock around both the read AND the write of the map

Reply:
Sorry for my previous comment. I was wrong, you're right. All good!